### PR TITLE
IBP-3039 Implement permissions for BMSAPI

### PR DESCRIPTION
**Pull request by darla-leafnode**

### CODE REVIEW

**Comment by darla-leafnode** [comment]
@mderamos - please apply code cleanup and formatter on all changed classes

File: src/main/java/org/ibp/api/java/impl/middleware/crop/CropServiceImpl.java [hunk 1]
```diff
@@ -13,6 +14,7 @@
 public class CropServiceImpl implements CropService {
 
 	@Autowired
+	@Lazy
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/crop/CropServiceImpl.java]: @darla-leafnode quick question, why is it @Lazy?

**Comment by clarysabel** [comment]
Hi @darla-leafnode 
I have reviewed the PR and seems fine to me. I left a question just for curiosity. 
Great job to all of you, I have been waiting for this improvement for a while :1st_place_medal:

**Comment by darla-leafnode** [comment]
Thanks @clarysabel for reviewing. 
Regarding use of @Lazy annotation, we used that because were experiencing failing integration tests because of circular dependency issue when autowiring:
`From MiddlewareFactory -> ContextResolver (autowired) -> ProgramService (autowired) -> WorkbenchDataManager(autowired).
` 
`WorkbenchDataManager `could not be autowired from such context because its bean initializer is in `MiddlewareFactory`.

**Pull request merged**